### ISSUE TITLE
Fault Management Subsystem

### ISSUE THREAD

**Issue by Firstyear** [kanidm/kanidm#2183]
Similar to the fault management tool in Solaris, it would be good to collect errors/faults into a single subsystem to allow review and resolution. This ties in heavily to #1984 which allows each unique error to be coalesced to a single location and then give precise outputs and actionable outputs.

**Comment by yaleman** [comment]
Is this an example of the kind of thing you're describing? Seems neat

https://docs.oracle.com/cd/E23824_01/html/821-1451/glisy.html

**Comment by Firstyear** [comment]
Yep, this is exactly what I'm thinking of. You can see it in ZFS too. In solaris it applied to every part of the system, which is just excellent.
